packages/credentials-interface/src/credentials.types.ts

```diff
@@ -1,11 +1,32 @@
-import { ICredentialSubject, IJsonLdCredential, IProof } from '@sphereon/pex';
+import {
+  ICredentialStatus,
+  ICredentialSubject,
+  IJsonLdCredential,
+  IProof,
+} from '@sphereon/pex';
 import { EIP712Proof } from './eip712proof.types';
 
 export type CredentialSubject = ICredentialSubject;
 
 export interface Credential<T extends CredentialSubject>
   extends IJsonLdCredential {
   credentialSubject: T;
+  credentialStatus?: StatusList2021Entry;
+}
+
+export enum CredentialStatusType {
+  StatusList2021Entry = 'StatusList2021Entry',
+}
+
+export enum CredentialStatusPurpose {
+  REVOCATION = 'revocation',
+}
+
+export interface StatusList2021Entry extends ICredentialStatus {
```

Comment:
`ICredentialStatus` interface require a `id`, in `StatusList2021Entry` we don't want that.

Reply:
I don't think this is proper use for never type. Besides since id is required `never` will not allow to ignore it

Reply:
This is a case that `never` was intruded. For example this is a Typescript implementation of exclude utility types `type Exclude<T, U> = T extends U ? never : T;`. We are extending object, so we can override every property that we want

Reply:
Sorry, we need `id` type, bc `DIDKit` is requiring it. So maybe id would be the as `statusListCredential`?

Reply:
specification doesn't allow `id` to be url of status list. perhaps it can be just uuid

Reply:
Can be `id` must be an URI - https://www.w3.org/TR/vc-data-model/#status. URL is a valid URI